itou/utils/widgets.py

```diff
@@ -117,3 +118,30 @@ def optgroups(self, name, value, attrs=None):
             subgroup = default[1]
             subgroup.append(self.create_option(name, option_value, option_label, selected_choices, index))
         return groups
+
+
+class AddressAutocompleteWidget(RemoteAutocompleteSelect2Widget):
+    # Be careful when using this widget because it requires the following fields to be included in the form:
+    #
+    # "address_line_1",
+    # "address_line_2",
+    # "post_code",
+    # "city",
+    # "insee_code",
+    # "latitude",
+    # "longitude",
+    # "geocoding_score",
+    # "ban_api_resolved_address",
+
+    class Media:
+        js = ["js/address_autocomplete_fields.js"]
+
+    def build_attrs(self, base_attrs, extra_attrs=None):
+        attrs = super().build_attrs(base_attrs, extra_attrs=extra_attrs)
+        attrs["data-ajax--url"] = f"{settings.API_BAN_BASE_URL}/search/"
+        attrs["data-ajax--cache"] = "true"
+        attrs["data-ajax--type"] = "GET"
+        attrs["data-minimum-input-length"] = 3
+        attrs["data-placeholder"] = "Ex. 102 Quai de Jemmapes 75010 Paris"
+
+        return attrs
```

Comment:
`cache = True` & `type = GET` sont déjà les valeurs par défaut: https://api.jquery.com/jQuery.ajax/

```suggestion
    def build_attrs(self, base_attrs, extra_attrs=None):
        return super().build_attrs(base_attrs, extra_attrs=extra_attrs) | {
            "data-ajax--url": f"{settings.API_BAN_BASE_URL}/search/",
            "data-minimum-input-length": 3,
            "data-placeholder": "Ex. 102 Quai de Jemmapes 75010 Paris",
        }
```